swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -234,39 +268,34 @@ service /covid/status on new http:Listener(9000) {
         return covidTable.toArray();
     }
 
-    resource function post countries(@http:Payload CovidEntry[] covidEntries)
-                                    returns CovidEntry[]|ConflictingIsoCodesError {
+    resource function post countries(CovidEntry[] covidEntries)
+            returns CovidEntry[]|IsoCodeConflict {
 
         string[] conflictingISOs = from CovidEntry covidEntry in covidEntries
             where covidTable.hasKey(covidEntry.iso_code)
             select covidEntry.iso_code;
 
         if conflictingISOs.length() > 0 {
             return {
-                body: {
-                    errmsg: string:'join(" ", "Conflicting ISO Codes:", ...conflictingISOs)
-                }
+                body: string:'join(" ", "Conflicting ISO Codes:", ...conflictingISOs)
             };
-        } else {
-            covidEntries.forEach(covdiEntry => covidTable.add(covdiEntry));
-            return covidEntries;
         }
+
+        covidEntries.forEach(covidEntry => covidTable.add(covidEntry));
+        return covidEntries;
     }
 
-    resource function get countries/[string iso_code]() returns CovidEntry|InvalidIsoCodeError {
-        CovidEntry? covidEntry = covidTable[iso_code];
-        if covidEntry is () {
-            return {
-                body: {
-                    errmsg: string `Invalid ISO Code: ${iso_code}`
-                }
-            };
+    resource function get countries/[string iso_code]() returns CovidEntry|IsoCodeNotFound {
+        if covidTable.hasKey(iso_code) {
+            return covidTable.get(iso_code);
         }
-        return covidEntry;
+        return {
+            body: string `Invalid ISO Code: ${iso_code}`
+        };
     }
 }
 
-public type CovidEntry record {|
+type CovidEntry record {|
     readonly string iso_code;
     string country;
     decimal cases;
```

Comment:
Aren't these ints now?

Reply:
Please compare the whole thing against what you get when following the steps.

Reply:
Changed to `int`.